src/SonarScanner.MSBuild.PreProcessor/PreProcessor.cs

```diff
@@ -97,6 +94,10 @@ private async Task<bool> DoExecute(ProcessedArgs localSettings)
             return false;
         }
 
+        var jreResolver = factory.CreateJreResolver(server);
```

Comment:
I tested this manually, and it would break if `server` was null, which makes sense.
I moved it under the nullability test of the server.